### ISSUE TITLE
CRD stacks.stacks.crossplane.io too long

### ISSUE THREAD

**Issue by ryanmoran** [crossplaneio/crossplane#698]
### What happened?
We are attempting to install crossplane from the helm master chart version `0.0.0-912.81a61f0`. We aren't running tiller, and so we used `helm template crossplane.tgz > crossplane.yml` to generate our resource yaml and then attempted to `kubectl apply -f crossplane.yml`. We get an error that the `stacks.stacks.crossplane.io` CRD is invalid.

Here is the entire output:
```sh
$ kubectl -n crossplane apply -f crossplane.yml
serviceaccount/crossplane unchanged
serviceaccount/stack-manager unchanged
customresourcedefinition.apiextensions.k8s.io/providers.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/replicationgroupclasses.cache.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/replicationgroups.cache.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/eksclusterclasses.compute.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/eksclusters.compute.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/rdsinstanceclasses.database.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/rdsinstances.database.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/s3bucketclasses.storage.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/s3buckets.storage.aws.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/providers.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/resourcegroups.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/redis.cache.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/redisclasses.cache.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/aksclusterclasses.compute.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/aksclusters.compute.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/mysqlservers.database.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/postgresqlservers.database.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/sqlserverclasses.database.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/accountclasses.storage.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/accounts.storage.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/containerclasses.storage.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/containers.storage.azure.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/redisclusterpolicies.cache.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/redisclusters.cache.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/kubernetesclusterpolicies.compute.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/kubernetesclusters.compute.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/resourceclasses.core.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/mysqlinstancepolicies.database.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/mysqlinstances.database.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/postgresqlinstancepolicies.database.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/postgresqlinstances.database.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/cloudmemorystoreinstanceclasses.cache.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/cloudmemorystoreinstances.cache.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/gkeclusterclasses.compute.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/gkeclusters.compute.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/cloudsqlinstanceclasses.database.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/cloudsqlinstances.database.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/providers.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/bucketclasses.storage.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/buckets.storage.gcp.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/stackrequests.stacks.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/bucketpolicies.storage.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/buckets.storage.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/kubernetesapplicationresources.workload.crossplane.io configured
customresourcedefinition.apiextensions.k8s.io/kubernetesapplications.workload.crossplane.io configured
clusterrole.rbac.authorization.k8s.io/crossplane unchanged
clusterrolebinding.rbac.authorization.k8s.io/crossplane unchanged
clusterrolebinding.rbac.authorization.k8s.io/crossplane-stack-manager unchanged
deployment.apps/crossplane unchanged
deployment.apps/crossplane-stack-manager unchanged
The CustomResourceDefinition "stacks.stacks.crossplane.io" is invalid: metadata.annotations: Too long: must have at most 262144 characters
```

### How can we reproduce it?
```sh
helm repo add crossplane-master https://charts.crossplane.io/master/
helm fetch crossplane-master/crossplane --version 0.0.0-912.81a61f0
helm template crossplane-0.0.0-912.81a61f0.tgz > crossplane.yml
kubectl apply -f crossplane.yml
```

### What environment did it happen in?
Crossplane version: 0.0.0-912.81a61f0

* Cloud provider: Google Cloud
* Kubernetes version: `version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.8-gke.10", GitCommit:"f53039cc1e5295eed20969a4f10fb6ad99461e37", GitTreeState:"clean", BuildDate:"2019-06-19T20:48:40Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}`
* Kubernetes distribution: GKE

**Comment by displague** [comment]
Thanks for filing the detailed report, @ryanmoran!

I'm able to reproduce this this behavior too.  It's rather unfortunate and something that https://github.com/crossplaneio/crossplane/issues/630 should prevent in the future.

```console
$ kubectl apply -f cluster/charts/crossplane/crds/stacks.crossplane.io_stacks.yaml
The CustomResourceDefinition "stacks.stacks.crossplane.io" is invalid: metadata.annotations: Too long: must have at most 262144 characters

$ kubectl create -f cluster/charts/crossplane/crds/stacks.crossplane.io_stacks.yaml
customresourcedefinition.apiextensions.k8s.io/stacks.stacks.crossplane.io created
```

The annotation being referenced is the one that `kubectl apply` creates with a copy of the input file.  To work around this, a `kubectl create -f https://raw.githubusercontent.com/crossplaneio/crossplane/master/cluster/charts/crossplane/crds/stacks.crossplane.io_stacks.yaml` should progress the installation.  I'm not sure about how the rest of the helm chart would proceed with this type of intervention.  It might need a _force_ option.

I'll dig into this to see if we've introduced a generation bug.  I can't imagine that the spec would naturally have exceeded this size boundary based on our recent changes.

**Comment by jbw976** [comment]
We recently updated to kubebuilder, controller-runtime, and controller-tools v2.  The Stacks CRD (formally called Extensions) has always embedded an upstream Kubernetes `apps.DeploymentSpec` and a `batch.JobSpec`.

It looks like there's a massive difference with the controller-tools v2 CRD generation that includes ALL the comments and everything for these big specs into the schema validation section now, which was not included in the v1 generation tooling.  The yaml has gone from ~150 lines to over ~10k 🤕 

v1 tooling (150 lines): https://github.com/crossplaneio/crossplane/blob/f5809ac49e7ad68e411a2b0f1b30a6675e1e2cb7/cluster/charts/crossplane/crds/extensions/v1alpha1/extension.yaml

v2 tooling (10k lines): https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/crds/stacks.crossplane.io_stacks.yaml

I wonder if there are some v2 CRD generation flags to not be so crazy verbose with the generated CRD schema validation, or if we need to open an upstream bug in [controller-tools](https://github.com/kubernetes-sigs/controller-tools/) and potentially pursue a fix ourselves.

To be clear about the repo steps, does this happen if you do a plain `helm install`? Or does it only happen when you do a `helm template` followed by `kubectl apply`?

**Comment by displague** [comment]
@ryanmoran I misread your original commands. I assumed you were using `helm-install`.  `helm-install` does not encounter this annotation issue. 

@jbw976 This is apparently related to https://github.com/kubernetes-sigs/kubebuilder/issues/906

**Comment by jbw976** [comment]
Ah good link @displague, that issue had the following controller-tools issue and potential fix to limit the size of the description fields.  The PR has been merged, so we could consider picking it up soon:
https://github.com/kubernetes-sigs/controller-tools/issues/288
https://github.com/kubernetes-sigs/controller-tools/pull/299

**Comment by displague** [comment]
I would hope that we don't have to give-up validation, but incase these `kubectl` and `controller-tools`/`controller-runtime` related changes don't address our needs I've opened: 
https://github.com/kubernetes-sigs/kubebuilder/issues/962

**Comment by ryanmoran** [comment]
FWIW, I was able to get crossplane installed by cutting the `stacks.stacks.crossplane.io` CRD out of the giant `crossplane.yml` file I generated with `helm template`. I created that CRD by itself using `kubectl create` and then re-ran `kubectl apply` with the `crossplane.yml` again and everything seems to be happy at the moment.

**Comment by displague** [comment]
`helm install  --name crossplane --namespace crossplane-system  cluster/charts/crossplane/` also works around the `kubectl apply` limitation.

**Comment by jbw976** [comment]
This should be fixed in master now thanks to #702 from @displague.  Thanks! 🚀
